test/test.jl

```diff
@@ -395,3 +395,9 @@ end
 end
 @test_throws ErrorException @testset "$(error())" begin
 end
+
+io = IOBuffer()
+@test (print(io, Base.Test.Error(:test_error, "woot", 5, backtrace())); 1) == 1
```

Comment:
A representative test (that I tried to make work first) causes the tests to fail because there is no way to recover from a `@test` fail inside a `@testset`.